src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs

```diff
@@ -10,18 +10,16 @@ namespace Microsoft.Extensions.Logging.Console
 {
     public class ConsoleLogger : ILogger
     {
-        // Writing to console is not an atomic operation in the current implementation and since multiple logger
-        // instances are created with a different name. Also since Console is global, using a static lock is fine.
-        private static readonly object _lock = new object();
         private static readonly string _loglevelPadding = ": ";
         private static readonly string _messagePadding;
         private static readonly string _newLineWithMessagePadding;
 
         // ConsoleColor does not have a value to specify the 'Default' color
         private readonly ConsoleColor? DefaultConsoleColor = null;
 
-        private IConsole _console;
+        internal ConsoleLoggerProcessor QueueProcessor;
         private Func<string, LogLevel, bool> _filter;
+        private IConsole _console;
```

Comment:
Why not just pass the `QueueProcessor` as a ctor argument and set the IConsole  on the `QueueProcessor` instead of on the message?